snap/local/transgui.desktop

```
@@ -3,7 +3,7 @@ Version=1.0
 Name=TransGUI
 GenericName=Transmission Remote GUI
 Comment=Transmission Remote GUI
-Exec=@bindir@/transgui
+Exec=@bindir@/transgui "%u"
```

Consider suggestion from @pbsds , should we just use `%U` here?

---

I dunno. It certainly works the way i have proposed it. Since i have tested it and use it this way since then on Suse and Ubuntu.
and the "Most down-stream distributions i've seen" ©pbsds seem to be only arch.

---

[Here](https://github.com/transmission-remote-gui/transgui/blob/47727a7af8e4840f405b1a8d2754b25a6abfec6f/main.pas#L1272-L1276) we see it supports multiple files/urls. Reading pascal an adventure.

---

If so, we should use `%U`, here's the spec I found:

https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html

Code | Description
-- | --
%u | A single URL. Local files may either be passed as 	      file: URLs or as file path.
%U | A list of URLs. 	      Each URL is passed as a separate argument to 	      the executable program. Local files may either be passed as 	      file: URLs or as file path.

@dsvi would you mind to revise it ;)

---

Look, button:
```suggestion
Exec=@bindir@/transgui "%U"
```

BTW, here is [nixpkgs](https://github.com/NixOS/nixpkgs/blob/release-21.11/pkgs/applications/networking/p2p/transgui/default.nix#L46) doing the same.

---

Is there a way to change it in pull request? i have no idea how.
Would probably be simpler to just add %U directly to the repo

---

@dsvi sure! You can edit it in the branch(`patch-1) of this pull request, the result will be updated here!

➡️ https://github.com/dsvi/transgui/blob/patch-1/snap/local/transgui.desktop

---

Amazing! Thanks Peter. My life will never be the same again!